src/hybrid/parse/hybrid_callbacks.c

```diff
@@ -0,0 +1,404 @@
+        QueryError_SetError(status, QUERY_EADDARGS, "Multiple PARAMS are not allowed. Parameters can be defined only once");
+        return;
+    }
+    
+    // Validate argument count (must be even for key-value pairs)
+    if (paramsArgs->argc == 0 || paramsArgs->argc % 2) {
+        QueryError_SetError(status, QUERY_EADDARGS, "Parameters must be specified in PARAM VALUE pairs");
+        return;
+    }
+
+    // Create parameter dictionary and populate
+    dict *params = Param_DictCreate();
+    if (!params) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, "Failed to create parameter dictionary");
+        return;
+    }
+    
+    size_t value_len;
+    int n = AC_NumArgs(paramsArgs);
+    for (int i = 0; i < n; i += 2) {
+        const char *param = AC_GetStringNC(paramsArgs, NULL);
+        const char *value = AC_GetStringNC(paramsArgs, &value_len);
+        
+        if (DICT_ERR == Param_DictAdd(params, param, value, value_len, status)) {
+            Param_DictFree(params);  // Cleanup on error
+            return;
+        }
+    }
+    
+    ctx->searchopts->params = params;
+}
+
+// DIALECT callback - implements EXACT original logic from lines 341-349
+void handleDialect(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    ctx->specifiedArgs |= SPECIFIED_ARG_DIALECT;
+}
+
+// FORMAT callback - implements EXACT original logic from lines 359-366
+void handleFormat(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    ctx->specifiedArgs |= SPECIFIED_ARG_FORMAT;
+    const char *format = *(char**)value;
+    QueryError *status = ctx->status;
+    ctx->specifiedArgs |= SPECIFIED_ARG_FORMAT;
+    if (strcasecmp(format, "STRING") == 0) {
+        *ctx->reqFlags &= ~QEXEC_FORMAT_EXPAND;
+    } else if (strcasecmp(format, "EXPAND") == 0) {
+        *ctx->reqFlags |= QEXEC_FORMAT_EXPAND;
+    }
+}
+
+// Helper function to ensure extended mode for aggregation operations
+static int ensureExtendedMode(uint32_t *reqflags, const char *name, QueryError *status) {
+    if (*reqflags & QEXEC_F_IS_SEARCH) {
+        QueryError_SetWithoutUserDataFmt(status, QUERY_EINVAL,
+                               "option `%s` is mutually exclusive with simple (i.e. search) options",
+                               name);
+        return 0;
+    }
+    REQFLAGS_AddFlags(reqflags, QEXEC_F_IS_AGGREGATE);
+    return 1;
+}
+
+// GROUPBY callback - implements EXACT original logic from parseGroupby
+void handleGroupby(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    ctx->specifiedArgs |= SPECIFIED_ARG_GROUPBY;
+    ArgsCursor *ac = (ArgsCursor*)value;
+    QueryError *status = ctx->status;
+
+    if (!ensureExtendedMode(ctx->reqFlags, "GROUPBY", status)) {
+        return;
+    }
+
+    const long long nproperties = AC_NumRemaining(ac);
```

Comment:
would it be better to have unsigned? Should we protect against potential overflows?